src/xenia/cpu/backend/x64/x64_seq_vector.cc

```diff
@@ -562,6 +571,20 @@ struct VECTOR_ADD
                 } else {
                   e.vpaddd(e.xmm1, src1, src2);
 
+                  if (e.IsFeatureEnabled(kX64EmitAVX512Ortho |
+                                         kX64EmitAVX512DQ)) {
+                    e.vmovdqa32(e.xmm3, src1);
+                    e.vpternlogd(e.xmm3, e.xmm1, src2, 0b00100100);
+                    Opmask saturate = e.k1;
+                    e.vpmovd2m(saturate, e.xmm3);
+
+                    e.vpsrad(e.xmm1 | saturate, e.xmm1, 31);
+                    e.vpxord(e.xmm1 | saturate, e.xmm1,
+                             e.GetXmmConstPtr(XMMSignMaskI32));
+                    e.vmovdqa32(dest, e.xmm1);
```

Comment:
This works fine. But it's not ideal. For some reason I have to operate upon `e.xmm1` and then move it into `dest` at the end with `vmovdqa32` when I could have just operated upon `dest` directly. But it seems to fail unit tests if I do that. This is still faster than the non-AVX512 route but needs some deeper investigation since it could save an additional instruction.

Reply:
This would probably do better as a TODO comment next to this code instead of as a comment in the PR?

Reply:
Currently going to look into it a bit longer before undrafting this PR.

Reply:
Good to go now. Turns out, in some cases, `src1` or `src2` can be the same as `dest`. Causing some overwriting to occur if writing to `dest` too early and then reading from `src1` or `src2` again.